src/Select.tsx

```diff
@@ -15,14 +15,23 @@ export type SelectProps = {
         React.SelectHTMLAttributes<HTMLSelectElement>,
         HTMLSelectElement
     >;
-    children: ReactNode;
     /** Default: false */
     disabled?: boolean;
     /** Default: "default" */
     state?: "success" | "error" | "default";
     /** The message won't be displayed if state is "default" */
     stateRelatedMessage?: ReactNode;
     style?: CSSProperties;
+    options: SelectOption[];
+    placeholder?: string;
+};
+
+export type SelectOption<T = string> = {
+    value: T | number | readonly string[] | undefined;
```

Comment:
The type here isn't great, `T` can be anything it gives the impression that our select chan handle arbitrary types...  
It's not the case, to my knoelage only string and number works...